src/utils/PrecompiledHeaders/PrecompiledHeaders.java

```diff
@@ -0,0 +1,136 @@
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public final class PrecompiledHeaders {
+
+    private static final Pattern DEPENDENCY_LINE_PATTERN = Pattern.compile("\\s*(\\S+.hpp)\\s*\\\\?");
+    private static final Pattern INCLUDE_PATTERN = Pattern.compile("^#\\s*include \"([^\"]+)\"$");
+    private static final String OBJS_PATH = "hotspot/variant-server/libjvm/objs";
+    private static final String PRECOMPILED_HPP_NAME = "precompiled.hpp";
+    private static final String PRECOMPILED_HPP_PATH = "src/hotspot/share/precompiled/" + PRECOMPILED_HPP_NAME;
+    private static final String INLINE_HPP_SUFFIX = ".inline.hpp";
+    private static final String HOTSPOT_SOURCE_PREFIX = "/jdk/src/hotspot/share/";
+
+    private PrecompiledHeaders() {
+        throw new UnsupportedOperationException("Instances not allowed");
+    }
+
+    public static void main(String[] args) throws IOException {
+        if (args.length < 2 || args.length > 3) {
+            System.err.println("Usage: min_inclusion_count build_root [jdk_root=.]");
+            System.exit(1);
+        }
+
+        int minInclusionCount = Integer.parseInt(args[0]);
+        Path buildRoot = Path.of(args[1]).toAbsolutePath();
+        if (!Files.isDirectory(buildRoot)) {
+            throw new IllegalArgumentException("build_root is not a directory: " + buildRoot);
+        }
+        Path jdkRoot = Path.of(args.length == 3 ? args[2] : ".").toAbsolutePath();
+        if (!Files.isDirectory(jdkRoot)) {
+            throw new IllegalArgumentException("jdk_root is not a directory: " + jdkRoot);
+        }
+
+        Path objsPath = buildRoot.resolve(OBJS_PATH);
+        if (!Files.isDirectory(objsPath)) {
+            throw new IllegalArgumentException("Could not find 'objs' directory: " + objsPath);
+        }
+
+        // Count inclusion times for each header
+        Map<String, Integer> occurrences;
+        try (Stream<Path> files = Files.list(objsPath)) {
+            occurrences = files
+                    .filter(file -> file.getFileName().toString().endsWith(".d"))
+                    .filter(Predicate.not(file -> file.getFileName().toString().startsWith("BUILD_LIBJVM")))
+                    .filter(Predicate.not(file -> file.getFileName().toString().contains(PRECOMPILED_HPP_NAME)))
+                    .flatMap(file -> {
+                        try {
+                            // The first line contains the object name
+                            return Files.lines(file).skip(1);
```

Comment:
Or maybe `return Files.lines(file).skip(1).distince()`?

Reply:
`distinct()` has some overhead, do we need it? Do we expect duplicate headers in a file?